dashboard/src/scenes/person/OutOfActiveList.jsx

```diff
@@ -46,36 +47,79 @@ const OutOfActiveList = ({ person }) => {
     }
   };
 
-  const setOutOfActiveList = async (updatedPerson) => {
-    updatedPerson.outOfActiveList = true;
+  const setOutOfActiveList = async (data) => {
+    if (data.team === "all") {
+      if (data.outOfActiveListDate && outOfBoundariesDate(data.outOfActiveListDate))
+        return toast.error("La date de sortie de file active est hors limites (entre 1900 et 2100)");
+      const historyEntry = {
+        date: new Date(),
+        user: user._id,
+        data: {
+          outOfActiveList: { newValue: true },
+          outOfActiveListReasons: { newValue: data.outOfActiveListReasons },
+          outOfActiveListDate: { newValue: data.outOfActiveListDate },
+        },
+      };
 
-    if (updatedPerson.outOfActiveListDate && outOfBoundariesDate(updatedPerson.outOfActiveListDate))
-      return toast.error("La date de sortie de file active est hors limites (entre 1900 et 2100)");
+      const [error] = await tryFetchExpectOk(async () =>
+        API.put({
+          path: `/person/${person._id}`,
+          body: await encryptPerson({
+            ...person,
+            ...data,
+            outOfActiveList: true,
+            history: [...(cleanHistory(person.history) || []), historyEntry],
+          }),
+        })
+      );
+      if (!error) {
+        await refresh();
+        toast.success(person.name + " est hors de la file active");
+      }
+    } else {
+      if (person.assignedTeams.length === 1) {
+        return toast.error("Une personne doit être suivie par au moins une équipe");
+      }
+      const historyEntry = {
+        date: new Date(),
+        user: user._id,
+        data: {
+          assignedTeams: {
+            oldValue: person.assignedTeams,
+            newValue: person.assignedTeams.filter((team) => team !== data.team),
+          },
```

Comment:
juste parce que j'ai mis quelques secondes à percuter...
si t'aimes pas ne prends pas ma suggestion, no offence :)

```suggestion
      const nextAssignedTeams = person.assignedTeams.filter((team) => team !== data.team);
      if (nextAssignedTeams.length < 1) {
        return toast.error("Une personne doit être suivie par au moins une équipe");
      }
      const historyEntry = {
        date: new Date(),
        user: user._id,
        data: {
          assignedTeams: {
            oldValue: person.assignedTeams,
            newValue: nextAssignedTeams,
          },
```